lib/sie_exporter.rb

```diff
@@ -0,0 +1,117 @@
+# encoding: utf-8
+
+require 'sie'
+
+# Exports activity years to SIE.
+class SieExporter
+  def initialize(activity_year)
+    @activity_year = activity_year
+  end
+
+  def call
+    Sie::Document.new(self).render
+  end
+
+  def program
+    'MAGE'
+  end
+
+  def program_version
+    `git rev-parse --short HEAD`.chomp
+  end
+
+  def generated_on
+    Date.today
+  end
+
+  def financial_years
+    [activity_year.starts..activity_year.ends]
+  end
+
+  def company_name
+    'Konglig Datasektionen'
+  end
+
+  def accounts
+    activity_year.accounts.map do |account|
+      { number: account.number, description: account.name }
+    end
+  end
+
+  def balance_account_numbers
+    activity_year.accounts.where('number < ?', 3000).pluck(:number)
+  end
+
+  def closing_account_numbers
+    activity_year.accounts.where('number >= ?', 3000).pluck(:number)
+  end
+
+  def balance_before(account_number, date)
+    ingoing_balance = activity_year.accounts.
+      where(number: account_number).
+      first!.ingoing_balance
+    result = VoucherRow.
+      joins(:voucher).
+      where('vouchers.activity_year_id' => activity_year.id).
```

Comment:
Align the operands of an expression in an assignment spanning multiple lines.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.